routes/stats.js

```diff
@@ -1,12 +1,245 @@
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/components/schemas/InternalErrorResponse'
+ */
 router.get("/user/me", authenticate, asyncHandler(getCurrentUserStats));
 
+/**
+ * @swagger
+ * /stats/user/{id}:
+ *   get:
+ *     summary: Retrieve statistics for a specific user
+ *     description: >
+ *       Returns statistics specific to a user identified by their ID, including metrics such as total projects, comments, votes, and average project lifespan.
+ *     tags: [Stats]
+ *     parameters:
+ *       - in: path
+ *         name: id
+ *         required: true
+ *         schema:
+ *           type: string
+ *           format: uuid
+ *         description: UUID of the user
+ *     responses:
+ *       200:
+ *         description: User statistics retrieved successfully
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/components/schemas/UserStatsResponse'
+ *       400:
+ *         description: Bad request (missing user ID)
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/components/schemas/ValidationErrorResponse'
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/components/schemas/InternalErrorResponse'
+ */
 router.get("/user/:id", asyncHandler(getUserStats));
 
 module.exports = router;
+
+/**
+ * @swagger
+ * components:
+ *   schemas:
+ *     GlobalStatsResponse:
+ *       type: object
+ *       properties:
+ *         success:
+ *           type: boolean
+ *           example: true
+ *         status:
+ *           type: string
+ *           example: success
+ *         statusCode:
+ *           type: integer
+ *           example: 200
+ *         message:
+ *           type: string
+ *           example: Success
+ *         data:
+ *           type: object
+ *           properties:
+ *             totalUsers:
+ *               type: string
+ *               example: "78"
+ *             totalComments:
+ *               type: string
+ *               example: "2"
+ *             totalProjects:
+ *               type: string
+ *               example: "26"
+ *             averageLifespan:
+ *               type: string
+ *               example: "7.0000000000000000"
+ *             mostCommonCause:
+ *               type: string
+ *               example: "Bored"
+ *             funeralsToday:
+ *               type: string
+ *               example: "0"
+ *             totalVotes:
+ *               type: string
+ *               example: "2"
+ *             votesToday:
+ *               type: string
+ *               example: "0"
+ *             averageEulogyLength:
+ *               type: string
+ *               example: "137.8846153846153846"
+ *             rageQuitRate:
+ *               type: string
+ *               example: "23.0769230769230769"
```

Comment:
Where does all these properties become strings?

```suggestion
 *             totalUsers:
 *               type: integer
 *               example: 78
 *             totalComments:
 *               type: integer
 *               example: 2
 *             totalProjects:
 *               type: integer
 *               example: 26
 *             averageLifespan:
 *               type: number
 *               format: float  
 *               example: 7.0000000000000000
 *             mostCommonCause:
 *               type: string
 *               example: "Bored"
 *             funeralsToday:
 *               type: integer
 *               example: 0
 *             totalVotes:
 *               type: integer
 *               example: 2
 *             votesToday:
 *               type: integer
 *               example: 0
 *             averageEulogyLength:
 *               type: number
 *               format: float
 *               example: 137.8846153846153846
 *             rageQuitRate:
 *               type: number
 *               format: float
 *               example: 23.0769230769230769
```